snapcraft/internal/project_loader/grammar_processing/_part_grammar_processor.py

```diff
@@ -67,6 +87,14 @@ def __init__(self, *, plugin, properties, project_options, repo):
         self._stage_package_grammar = getattr(plugin, 'stage_packages', [])
         self.__stage_packages = set()
 
+    def get_source(self):
+        if not self.__source:
+            value = self._property_grammar['source']
+            self.__source = next(iter(grammar.process_grammar(
+                value if isinstance(value, list) else {value},
+                self._project_options, True)))
```

Comment:
There are no tests using `try`. They would fail. This needs to be a callable, as the docs say.

Reply:
Fixed.